src/utils.ts

```diff
@@ -7,125 +16,157 @@ import defaultLeadLayout from "./resources/default_layout/lead.json";
-                
-                return layout.view;
+        .otherwise(() => {
+          return match([curr.errorCode, curr.fields])
+            .with(["REQUIRED_FIELD_MISSING", ["ActivityDateTime"]], () => {
+              return `Please set a start date.\n\n`;
             })
-        .with(["Opportunity", "list"], () => {
-            const layout: OpportunityLayout = settings?.mapping_opportunity
-            ? JSON.parse(settings.mapping_opportunity)
-            : defaultOpportunityLayout
-            ;
-            
-            return layout.list;
-        })
-        .with(["Opportunity", "view"], () => {
-            const layout: OpportunityLayout = settings?.mapping_opportunity
-            ? JSON.parse(settings.mapping_opportunity)
-            : defaultOpportunityLayout
-            ;
-            
-            return layout.view;
-        })
-        .with(["Lead", "home"], () => {
-            const layout: LeadLayout = settings?.mapping_lead
-            ? JSON.parse(settings.mapping_lead)
-            : defaultLeadLayout
-            ;
-
-            return layout.home;
-        })
-        .with(["Lead", "view"], () => {
-            const layout: LeadLayout = settings?.mapping_lead
-            ? JSON.parse(settings.mapping_lead)
-            : defaultLeadLayout
-            ;
-            
-            return layout.view;
-        })
-        .with(["Account", "home"], () => {
-            const layout: AccountLayout = settings?.mapping_account
-            ? JSON.parse(settings.mapping_account)
-            : defaultAccountLayout
-            ;
-            
-            return layout.home;
-        })
-        .with(["Account", "view"], () => {
-            const layout: AccountLayout = settings?.mapping_account
-            ? JSON.parse(settings.mapping_account)
-            : defaultAccountLayout
-            ;
-            
-            return layout.view;
-        })
-        .with(["Task", "list"], () => {
-            const layout: TaskLayout = settings?.mapping_task
-            ? JSON.parse(settings.mapping_task)
-            : defaultTaskLayout;
-    
-            return layout.list;
-        })
-        .with(["Task", "view"], () => {
-            const layout: TaskLayout = settings?.mapping_task
-            ? JSON.parse(settings.mapping_task)
-            : defaultTaskLayout;
-    
-            return layout.list;
-        })
-        .with(["Event", "list"], () => {
-            const layout: EventLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultEventLayout
-            ;
-    
-            return layout.list;
-        })
-        .with(["Event", "view"], () => {
-            const layout: EventLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultEventLayout
-            ;
-    
-            return layout.list;
+            .with(["REQUIRED_FIELD_MISSING", ["DurationInMinutes"]], () => {
```

Comment:
Hard coded field, `DurationInMinutes`? what happens if someone changes are removes this field in Salesforce? Is that even possible? If it's not possible to remove/change this field  in Salesforce then this is *probably* fine.

Reply:
same answer as below